src/gui/providers/ogr/qgsogrsourceselect.cpp

```diff
@@ -556,4 +629,156 @@ void QgsOgrSourceSelect::showHelp()
+      continue;
+
+    const char *pszOptionName = CPLGetXMLValue( psItem, "name", nullptr );
+    if ( pszOptionName == nullptr )
+      continue;
+
+    // Exclude options that are not of vector scope
+    const char *pszScope = CPLGetXMLValue( psItem, "scope", nullptr );
+    if ( pszScope != nullptr && strstr( pszScope, "vector" ) == nullptr )
+      continue;
+
+    // The GPKG driver list a lot of options that are only for rasters
+    if ( bIsGPKG && strstr( pszOpenOptionList, "scope=" ) == nullptr &&
+         !EQUAL( pszOptionName, "LIST_ALL_TABLES" ) &&
+         !EQUAL( pszOptionName, "PRELUDE_STATEMENTS" ) )
+      continue;
+
+    // Do not list database options already asked in the database dialog
+    if ( radioSrcDatabase->isChecked() &&
+         ( EQUAL( pszOptionName, "USER" ) ||
+           EQUAL( pszOptionName, "PASSWORD" ) ||
+           EQUAL( pszOptionName, "HOST" ) ||
+           EQUAL( pszOptionName, "DBNAME" ) ||
+           EQUAL( pszOptionName, "DATABASE" ) ||
+           EQUAL( pszOptionName, "PORT" ) ||
+           EQUAL( pszOptionName, "SERVICE" ) ) )
+    {
+      continue;
+    }
+
+    const char *pszType = CPLGetXMLValue( psItem, "type", nullptr );
+    QStringList options;
+    if ( pszType && EQUAL( pszType, "string-select" ) )
+    {
+      for ( auto psOption = psItem->psChild; psOption != nullptr; psOption = psOption->psNext )
+      {
+        if ( psOption->eType != CXT_Element ||
+             !EQUAL( psOption->pszValue, "Value" ) ||
+             psOption->psChild == nullptr )
+        {
+          continue;
+        }
+        options << psOption->psChild->pszValue;
+      }
+    }
+
+    QLabel *label = new QLabel( pszOptionName );
+    QWidget *control = nullptr;
+    if ( pszType && EQUAL( pszType, "boolean" ) )
+    {
+      QComboBox *cb = new QComboBox();
+      cb->addItem( tr( "Yes" ), "YES" );
+      cb->addItem( tr( "No" ), "NO" );
+      cb->addItem( tr( "<Default>" ), QVariant( QVariant::String ) );
+      int idx = cb->findData( QVariant( QVariant::String ) );
+      cb->setCurrentIndex( idx );
+      control = cb;
+    }
+    else if ( !options.isEmpty() )
+    {
+      QComboBox *cb = new QComboBox();
+      for ( const QString &val : qgis::as_const( options ) )
+      {
+        cb->addItem( val, val );
+      }
+      cb->addItem( tr( "<Default>" ), QVariant( QVariant::String ) );
+      int idx = cb->findData( QVariant( QVariant::String ) );
+      cb->setCurrentIndex( idx );
+      control = cb;
+    }
+    else
+    {
+      QLineEdit *le = new QLineEdit( );
+      control = le;
+    }
+    control->setObjectName( pszOptionName );
+    mOpenOptionsWidgets.push_back( control );
+
+    const char *pszDescription = CPLGetXMLValue( psItem, "description", nullptr );
+    if ( pszDescription )
+    {
+      label->setToolTip( QStringLiteral( "<p>%1</p>" ).arg( pszDescription ) );
+      control->setToolTip( QStringLiteral( "<p>%1</p>" ).arg( pszDescription ) );
```

Comment:
Hi, this PR introduces a number of strings in the dialog. These tooltips are helpful but is there a chance they could be made translatable, please? See https://lists.osgeo.org/pipermail/qgis-tr/2024-January/001387.html
Thanks.

Reply:
There is no immediate straightforward way of making them translatable as that would require to extract them from GDAL first. I guess that could potentially be done, and I'm CC'ing @elpaso if he might be interested in looking that on GDAL side.
I would imagine a Python script querying the drivers, getting the XML of their creation options and extracting the "description" attribute to create a gettext .pot file
The main question to me is who would translate that then. I don't think GDAL itself has a sufficiently large i18n interested community to tackle that task (chicken & egg problem as there's no translation mechanism in GDAL currently), so maybe we should rely on QGIS community to do that.
So the process would be:
1. GDAL script creating .pot file
2. Importing that into QGIS and create/refresh associated .po file
3. Have QGIS community translate those GDAL .po files
4. Get the translated files (not sure how Transiflex works, but is it possible to get an individual file back?) and re-inject into GDAL, and add a gettext-based mechanism in GDAL to get translated strings

Step 4 is probably optional for a QGIS-only context, but would probably be needed to justify using GDAL sponsored time on that task (we'd want also other frontends to GDAL than QGIS to be able to benefit from translated strings).

Behind the technicalities, there's also a licensing issue. What is the license of translated files done in the QGIS context? Ideally, we'd want them to be usable under a liberal license, compatible of the GDAL MIT license if we re-inject them into GDAL.